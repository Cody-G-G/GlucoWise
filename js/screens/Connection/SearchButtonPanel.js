'use strict';
import React, {Component} from 'react'; // this is from a 3rd party dependency NPM module, "react"
import {Icon} from 'native-base'; // this is from a 3rd party dependency NPM module, "native-base"
import {View, TouchableOpacity, Text} from 'react-native'; // this is from a 3rd party dependency NPM module, "react-native"
import styles from "./styles";

export default class SearchButtonPanel extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.searchButtonPanel}>
                <TouchableOpacity onPress={this.props.onPress} style={styles.searchButton}>
                    <Icon style={styles.searchButtonIcon} theme={{iconFamily: "MaterialIcons"}} name="bluetooth"/>
                    <Text style={styles.searchButtonText}>Search Devices</Text>
                </TouchableOpacity>
            </View>
        );
    }
}