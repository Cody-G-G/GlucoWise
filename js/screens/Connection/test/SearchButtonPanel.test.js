'use strict';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import SearchButtonPanel from "../SearchButtonPanel"

test('SearchButtonPanel - renders correctly', () => {
    const tree = renderer.create(
        <SearchButtonPanel/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
