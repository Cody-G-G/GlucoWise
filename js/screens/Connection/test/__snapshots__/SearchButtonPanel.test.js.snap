// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButtonPanel - renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 20,
      "paddingTop": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "royalblue",
        "borderColor": "#5067FF",
        "borderRadius": 5,
        "borderWidth": 0,
        "elevation": 2,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-around",
        "opacity": 1,
        "paddingHorizontal": 8,
        "paddingVertical": 6,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#fff",
            "fontSize": 45,
            "lineHeight": 52,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      theme={
        Object {
          "iconFamily": "MaterialIcons",
        }
      }
    >
      î†§
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "HelveticaNeue",
          "fontSize": 22.5,
          "lineHeight": 29,
          "marginLeft": 7,
          "marginRight": 0,
        }
      }
    >
      Search Devices
    </Text>
  </View>
</View>
`;
